dags/inference/configs/maxdiffusion_inference_gce_config.py

```diff
@@ -0,0 +1,109 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to construct configs for maxdiffusion inference DAG."""
+
+import json
+from typing import Dict
+from xlml.apis import gcp_config, metric_config, task, test_config
+from dags import test_owner
+from dags.multipod.configs import common
+from dags.vm_resource import TpuVersion, Project, RuntimeVersion
+
+PROJECT_NAME = Project.CLOUD_ML_AUTO_SOLUTIONS.value
+RUNTIME_IMAGE = RuntimeVersion.TPU_UBUNTU2204_BASE.value
+GCS_SUBFOLDER_PREFIX = test_owner.Team.INFERENCE.value
+
+
+def get_maxdiffusion_inference_nightly_config(
+    tpu_version: TpuVersion,
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    test_name: str,
+    test_mode: common.SetupMode,
+    project_name: str = PROJECT_NAME,
+    runtime_version: str = RUNTIME_IMAGE,
+    network: str = "default",
+    subnetwork: str = "default",
+    is_tpu_reserved: bool = True,
+    num_slices: int = 1,
+    model_configs: Dict = {},
+) -> task.TpuQueuedResourceTask:
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.BENCHMARK_DATASET,
+  )
+
+  set_up_cmds = (
+      "pip install --upgrade pip",
+      # Download maxdiffusion
+      "git clone -b inference_utils https://github.com/google/maxdiffusion.git",
+      # Create a python virtual environment
+      "sudo apt-get -y update",
+      "sudo apt-get -y install python3.10-venv",
+      "python -m venv .env",
+      "source .env/bin/activate",
+      # Setup Maxdiffusion
+      "cd maxdiffusion",
+      "pip3 install jax[tpu] -f https://storage.googleapis.com/jax-releases/libtpu_releases.html",
+      "pip3 install -r requirements.txt",
+      "pip3 install ."
+  )
+
+  additional_metadata_dict = {
+      "per_device_batch_size": f"{model_configs['per_device_batch_size']}",
+  }
+
+  run_model_cmds = (
+      # Start virtual environment
+      "source .env/bin/activate",
+      ### Benchmark
+      "cd maxdiffusion",
+      # Configure flags
+      """ python -m src.maxdiffusion.generate_sdxl src/maxdiffusion/configs/base_xl.yml run_name="my_run" """,
+      # Give server time to start
+      f"sleep {model_configs['sleep_time']}",
+      f"gsutil cp metrics.json {metric_config.SshEnvVars.GCS_OUTPUT.value}",
+  )
+
+  job_test_config = test_config.TpuVmTest(
+      test_config.Tpu(
+          version=tpu_version,
+          cores=tpu_cores,
+          runtime_version=runtime_version,
+          reserved=is_tpu_reserved,
+          network=network,
+          subnetwork=subnetwork,
+      ),
+      test_name=test_name,
+      set_up_cmds=set_up_cmds,
+      run_model_cmds=run_model_cmds,
+      time_out_in_min=time_out_in_min,
```

Comment:
```suggestion
      timeout=datetime.timedelta(minutes=time_out_in_min),
```
Due to https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/289

Reply:
done